pyproject.toml

```diff
@@ -96,26 +96,26 @@ filterwarnings = ["error"]
 
 [tool.coverage.run]
 omit = [
-    "venv/*", 
-    "httpcore/_sync/*"
+    "venv/*",
+    "httpcore/_sync/*",
 ]
-include = ["httpcore/*", "tests/*"]
-
-[tool.ruff]
-exclude = [
-    "httpcore/_sync",
-    "tests/_sync",
+include = [
+    "httpcore/*",
+    "tests/*",
 ]
-line-length = 88
+
+[tool.ruff.lint]
 select = [
     "E",
     "F",
     "W",
-    "I"
+    "I",
 ]
 
-[tool.ruff.pycodestyle]
-max-line-length = 120
-
-[tool.ruff.isort]
+[tool.ruff.lint.isort]
 combine-as-imports = true
+
+[tool.ruff.lint.per-file-ignores]
+# needed for unasync.py
+"httpcore/_sync/*" = ["I001"]
+"tests/_sync/*" = ["I001"]
```

Comment:
Do we really need to avoid excluding sync directories? When we need to format/lint them? they are auto generated.

Reply:
They are pre-generated. it means currently, they are already available in sdist, wheels and repository.
It doesn't look good that readers navigate through unformatted sources after installation.
For example, after install the `httpcore`, I'll navigate to `_sync` folder's contents and open a single file, then I'll see linter/formatter warnings in my IDE.

Reply:
They are pre-generated. it means currently, they are already available in sdist, wheels and repository.
It doesn't look good that readers navigate through unformatted sources after installation.
For example, after installation, I'll navigate to `site-packages/httpcore` then in `_sync` folder's contents I open a single file, then I'll see linter/formatter warnings in my IDE.

Reply:
When it's possible to have them, why excluding them?

Reply:
> we don't know how many additional rules may be broken when introducing a new feature. We need add new rules that break sync files into the configuration every time, which makes the maintenance process more difficult.

The codebase is almost stable, I think the only violation rule would be import sorting violation. by checking `unasync.py` script I don't except new violation happening when adding new feature other than I001 (and F811 which can be in-line ignoring this without changing `pyproject.toml`)

IMO, we could give it a try, since always doable in future to revert this change.